blog/2024-03-01-modern-angular-bm/README.md

```diff
@@ -0,0 +1,424 @@
+
+Als nächstes Wollen wir uns den Interceptor zur Authentifizierung ansehen.
+Dieser ist bisher als Klasse implementiert und sieht wie folgt aus:
+
+```ts
+// ...
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  private authService = inject(AuthService);
+
+  intercept(
+    request: HttpRequest<unknown>,
+    next: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    const token = '1234567890';
+
+    if (this.authService.isAuthenticated) {
+      // ...
+      return next.handle(reqWithToken);
+    } else {
+      return next.handle(request);
+    }
+  }
+}
+```
+
+Bei Verwendung eines functional Interceptors legen wir eine Factory Funktion an, die den Request und eine `HttpHandlerFn` übergeben bekommt.
+Die `HttpHandlerFn` ist equivalent zum Aufruf von `.handle` vom zuvor verwendeten `HttpHandler`.
+
+```ts
+import { HttpEvent, HttpHandlerFn, HttpRequest } from '@angular/common/http';
+// ...
+
+export function authInterceptor(
+  req: HttpRequest<unknown>,
+  next: HttpHandlerFn
+): Observable<HttpEvent<unknown>> {
+  const authService = inject(AuthService)
+  const token = '1234567890';
+
+  if (authService.isAuthenticated) {
+    // ...
+    return next(reqWithToken);
+  } else {
+    return next(req);
+  }
+}
+```
+
+Jetzt, da wir keine Klasse mehr verwenden, die das DI Token `HTTP_INTERCEPTORS` überlädt, müssen wir noch eine Anpassung in unserem `main.ts` vornehmen.
+Hier nutzen wir nun `withInterceptors()` statt `withInterceptorsFromDi()` und übergeben den Interceptor im Array.
+Entsprechend können wir das Token-Konfigurationsobjekt mit dem Verweis auf die Klasse entfernen.
+Bei der Gelegenheit können wir auch `provideHttpClient()` zusätzlich die Funktion `withFetch()` mit übergeben, um Angular zur Nutzung der moderneren [Fetch-API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) zu bewegen.
+
+```ts
+import {
+  provideHttpClient,
+  withInterceptors,
+  withFetch
+} from '@angular/common/http';
+// ...
+import { authInterceptor } from './app/shared/auth.interceptor';
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    // ...
+    provideHttpClient(
+      withFetch(),
+      withInterceptors([authInterceptor])
+    ),
+  ]
+}).catch(err => console.error(err));
+```
+
+## Control Flow
+
+Mit Angular 17 kam der neue Control Flow ins Spiel.
+Dieser erlaubt es uns Verzweigungen und Schleifen mit `@if()` und `@for()` direkt im Template von Angular Komponenten zu nutzen,
+ohne zuvor passende Direktiven dafür importieren zu müssen.
+
+Auch wenn der Control Flow mit Angular 17 noch im Developer Preview ist, empfehlen wir die Nutzung bereits jetzt.
+
+Um nicht alle Komponenten von Hand migrieren zu müssen, gibt uns Angular auch hier ein Migrationswerkzeug mit an die Hand.
+Der Aufruf von `ng generate @angular/core:control-flow` führt die Migration der Template Syntax durch und entfernt im Nachgang auch automatisch die unnötigen Importe der Direktiven `NgIf` und `NgForOf`.
```

Comment:
Befehl lieber in einen Codeblock, der liest sich im FLießtext seltsam

Reply:
`NgForOf` ist zwar der richtige Name, aber üblicherweise importiert man ja `NgFor`.